spec/integration/user/show_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :system do
+  describe 'profile page' do
+    it 'I can see the user`s profile picture.' do
+      visit '/users/108'
+      expect(page).to have_selector('img')
+    end
+
+    it 'I can see the user`s username' do
+      visit '/users/108'
+      expect(page).to have_content('test')
+    end
+
+    it 'I can see the number of posts the user has written' do
+      visit '/users/108'
+      expect(page).to have_content('Number of posts:')
+    end
+
+    it 'I can see the user`s bio' do
+      visit '/users/108'
+      expect(page).to have_content('Bio')
+    end
+
+    it 'I can see the user`s last 3 posts' do
+      visit '/users/108'
+      expect(page).to have_selector('a')
+      expect(page).to have_selector('a')
+      expect(page).to have_selector('a')
+    end
```

Comment:
- [ ] Kindly rewrite this test for checking the first three posts and ensure it is more thorough. There is no indication that these assertions list the three posts. You are just selecting any item on the page with the "<a>" tag which can be anything really. I would recommend you use the actual content of the posts themselves like their individual post titles.